File: src/Kattis/COMP321/A2/B.java
package Kattis.COMP321.A2;

import java.util.LinkedList;
import java.util.Scanner;

public class B {
    public static void main(String[] args) {
        System.out.println(backspace());
    }

    private static String backspace() {
        // Get input
        Scanner scanner = new Scanner(System.in);
        String input = scanner.nextLine();
        StringBuilder output = new StringBuilder();

        // Format input
        LinkedList<Character> outputList = new LinkedList<>();
        int size = 0; // To avoid computing outputList.size() many times
        char cur;
        for (int i = 0; i < input.length(); i++) {
            cur = input.charAt(i);
            if (cur == '<' && size > 0) {
                outputList.removeLast();
                size--;
            } else if (cur != '<') {
                outputList.addLast(cur);
                size++;
            }
        }

        // Create output
        for (char c: outputList) {
            output.append(c);
        }

        return output.toString();
    }
}

File: src/Kattis/COMP321/A2/B.cpp
#include <iostream>

using namespace std;

int main() {
    // Declare vars
    string input;
    char output[1000000];
    int index = 0;
    char c;

    // Get input
    cin >> input;

    // Format input
    for (int i = 0; i < input.length(); i++) {
        c = input.at(i);
        if (c >= 'a') {
            output[index++] = c;
        } else {
            index--;
        }
    }

    // Print output
    for (int i = 0; i < index; i++) {
        cout << output[i];
    }

    // Exit
    return 0;
}
